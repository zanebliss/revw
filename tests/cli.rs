use std::{
    env::set_current_dir,
    path::Path,
    process::Command,
};

use assert_cmd::Command as AssertCmd;
use common::restore_root_dir;
use tempfile::tempdir;

mod common;

// #[test]
// fn sanity_check_with_rspec() -> Result<(), Box<dyn std::error::Error>> {
//     // create a temp directory
//     // cd into it
//     // copy files from fixture directories into it
//     // execute the setup script
//     let tmp_dir = tempdir()?;
//     let path = Path::new("tests/fixtures/sanity_check_with_rspec/setup.sh");
//     Command::new(path).arg(tmp_dir.path()).output()?;
//     set_current_dir(tmp_dir)?;

//     let mut cmd = assert_cmd::Command::cargo_bin("teva")?;

//     insta::with_settings!({ filters => vec![
//         (r"\b\d*\.\d+\b", "[TIME]"),
//         (r"33m[a-z0-9]{7}", "[SHA]")
//     ]}, {
//         insta::assert_snapshot!(String::from_utf8(cmd.output().unwrap().stdout).unwrap());
//     });

//     restore_root_dir()?;

//     Ok(())
// }

#[test]
fn not_enough_commits() -> Result<(), Box<dyn std::error::Error>> {
    let tmp_dir = tempdir()?;
    let setup_script = Path::new("tests/fixtures/not_enough_commits/setup.sh");
    println!("{:?}", String::from_utf8(Command::new("ls").arg("tests/fixtures/not_enough_commits").output().unwrap().stdout));
    // Command::new(setup_script).arg(tmp_dir.path()).output()?;
    // set_current_dir(tmp_dir.path())?;

    let mut cmd = AssertCmd::cargo_bin("teva")?;

    insta::with_settings!({ filters => vec![
        (r"\b\d*\.\d+\b", "[TIME]"),
        (r"33m[a-z0-9]{7}", "[SHA]")
    ]}, {
        insta::assert_snapshot!(String::from_utf8(cmd.output().unwrap().stdout).unwrap());
    });

    restore_root_dir()?;

    Ok(())
}
