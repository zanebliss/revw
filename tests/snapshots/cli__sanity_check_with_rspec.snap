---
source: tests/cli.rs
expression: "String::from_utf8(cmd.output().unwrap().stdout).unwrap()"
---
 _________  _______   ___      ___ ________     
|\___   ___\\  ___ \ |\  \    /  /|\   __  \    
\|___ \  \_\ \   __/|\ \  \  /  / | \  \|\  \   
     \ \  \ \ \  \_|/_\ \  \/  / / \ \   __  \  
      \ \  \ \ \  \_|\ \ \    / /   \ \  \ \  \ 
       \ \__\ \ \_______\ \__/ /     \ \__\ \__\
        \|__|  \|_______|\|__|/       \|__|\|__|
    
[94m[TEVA][0m ‚öôÔ∏è Setting up environment... Done ‚úîÔ∏è
[94m[TEVA][0m
[94m[TEVA][0m [[SHA][0m add bar (1 of 3)
[94m[TEVA][0m Changed files: bar.rb spec/bar_spec.rb
[94m[TEVA][0m Running tests...
.

Finished in [TIME] seconds (files took [TIME] seconds to load)
1 example, 0 failures

[94m[TEVA][0m [[SHA][0m add baz (2 of 3)
[94m[TEVA][0m Changed files: baz.rb spec/baz_spec.rb
[94m[TEVA][0m Running tests...
.F

Failures:

  1) Baz is expected to eq 2
     Failure/Error: expect(1).to eq(2)

       expected: 2
            got: 1

       (compared using ==)
     # ./spec/baz_spec.rb:6:in `block (2 levels) in <top (required)>'

Finished in [TIME] seconds (files took [TIME] seconds to load)
2 examples, 1 failure

Failed examples:

rspec ./spec/baz_spec.rb:5 # Baz is expected to eq 2

[94m[TEVA][0m [[SHA][0m add fizz (3 of 3)
[94m[TEVA][0m Changed files: fizz.rb spec/fizz_spec.rb
[94m[TEVA][0m Running tests...
.F.

Failures:

  1) Baz is expected to eq 2
     Failure/Error: expect(1).to eq(2)

       expected: 2
            got: 1

       (compared using ==)
     # ./spec/baz_spec.rb:6:in `block (2 levels) in <top (required)>'

Finished in [TIME] seconds (files took [TIME] seconds to load)
3 examples, 1 failure

Failed examples:

rspec ./spec/baz_spec.rb:5 # Baz is expected to eq 2
